src/pip/_internal/utils/logging.py

```diff
@@ -115,6 +115,13 @@ def format(self, record):
 
         return msg
 
+    def flush(self):
+        """Ignore `BrokenPipeErrors` on `flush()`."""
+        try:
+            self.flush()
+        except BrokenPipeError:
```

Comment:
BrokenPipeError wasn't defined in Python 2, so you'll need something different for that case